File: qqbot-plugin-testkit/src/main/java/com/mieai/qqbot/plugin/testkit/FakePluginLogger.java
package com.mieai.qqbot.plugin.testkit;

import com.mieai.qqbot.plugin.api.PluginLogger;
import java.util.ArrayList;
import java.util.List;

/** Recording logger that never writes test messages to the process log. */
public final class FakePluginLogger implements PluginLogger {
    private final List<Entry> entries = new ArrayList<>();

    @Override public synchronized void info(String message) { entries.add(new Entry("INFO", message, null)); }
    @Override public synchronized void warn(String message) { entries.add(new Entry("WARN", message, null)); }
    @Override public synchronized void error(String message, Throwable cause) { entries.add(new Entry("ERROR", message, cause)); }

    public synchronized List<Entry> entries() { return List.copyOf(entries); }

    public record Entry(String level, String message, Throwable cause) {}
}

File: qqbot-plugin-testkit/src/main/kotlin/com/mieai/qqbot/plugin/testkit/FakePluginLogger.kt
package com.mieai.qqbot.plugin.testkit

import com.mieai.qqbot.plugin.api.PluginLogger

/** Recording logger that never writes test messages to the process log. */
class FakePluginLogger : PluginLogger {
    private val entries = mutableListOf<Entry>()

    @Synchronized
    override fun info(message: String) {
        entries.add(Entry("INFO", message, null))
    }

    @Synchronized
    override fun warn(message: String) {
        entries.add(Entry("WARN", message, null))
    }

    @Synchronized
    override fun error(message: String, cause: Throwable) {
        entries.add(Entry("ERROR", message, cause))
    }

    @Synchronized
    fun entries(): List<Entry> = entries.toList()

    data class Entry(
        val level: String,
        val message: String,
        val cause: Throwable?,
    )
}
